File: backend/HealthEco.Core/Entities/Doctor.cs
// 2. Sửa Doctor và các entities khác nếu cần
// HealthEco.Core/Entities/Doctor.cs - CẬP NHẬT
using System.Collections.Generic;

namespace HealthEco.Core.Entities
{
    public class Doctor : BaseEntity
    {
        public int UserId { get; set; }
        public User User { get; set; } = null!;

        public string MedicalLicense { get; set; } = null!;
        public string? LicenseImageUrl { get; set; }

        public int? SpecializationId { get; set; }
        public Specialization? Specialization { get; set; }

        public int YearsExperience { get; set; } = 0;
        public string? Qualifications { get; set; }
        public string? Bio { get; set; }

        public decimal ConsultationFee { get; set; } = 0;
        public decimal Rating { get; set; } = 0.00m;
        public int TotalReviews { get; set; } = 0;
        public bool IsVerified { get; set; } = false;

        public ICollection<Appointment> Appointments { get; set; } = new List<Appointment>();
        public ICollection<Review> Reviews { get; set; } = new List<Review>();
    }
}

File: frontend/src/types/doctor.ts
// frontend/src/types/doctor.ts
export interface Specialization {
    id: number;
    name: string;
    description: string;
    iconUrl: string;
    isActive: boolean;
}

export interface Doctor {
    id: number;
    userId: number;
    medicalLicense: string;
    licenseImageUrl: string;
    specializationId: number;
    specialization?: Specialization;
    yearsExperience: number;
    qualifications: string;
    bio: string;
    consultationFee: number;
    rating: number;
    totalReviews: number;
    isVerified: boolean;
    createdAt: string;
    updatedAt: string;
    user: {
        id: number;
        email: string;
        fullName: string;
        role: string;
        phoneNumber?: string;
        dateOfBirth?: string;
        address?: string;
        city?: string;
        avatarUrl?: string;
        isActive: boolean;
        isEmailVerified: boolean;
        createdAt: string;
        updatedAt?: string;
    };
}

export interface DoctorRegisterRequest {
    email: string;
    password: string;
    fullName: string;
    phoneNumber?: string;
    dateOfBirth?: string | null;
    address?: string;
    city?: string;
    medicalLicense: string;
    licenseImageUrl?: string;
    specializationId: number;
    yearsExperience: number;
    qualifications: string;
    bio?: string;
    consultationFee: number;
}

export interface DoctorUpdateRequest {
    medicalLicense?: string;
    licenseImageUrl?: string;
    specializationId?: number;
    yearsExperience: number;
    qualifications?: string;
    bio?: string;
    consultationFee: number;
}

export interface DoctorSearchParams {
    searchTerm?: string;
    specializationId?: number;
    city?: string;
    minFee?: number;
    maxFee?: number;
    isVerified?: boolean;
    page?: number;
    pageSize?: number;
    sortBy?: string;
    sortDescending?: boolean;
}
export interface SpecializationResponse {
    id: number;
    name: string;
    description: string;
    iconUrl: string;
    isActive: boolean;
}
export interface DoctorResponse {
    id: number;
    userId: number;
    medicalLicense: string;
    licenseImageUrl?: string;
    specializationId?: number;
    specialization?: SpecializationResponse;
    yearsExperience: number;
    qualifications: string;
    bio?: string;
    consultationFee: number;
    rating: number;
    totalReviews: number;
    isVerified: boolean;
    createdAt: string;
    updatedAt: string;
    user: UserResponse;
}
export interface UserResponse {
    id: number;
    email: string;
    fullName: string;
    role: string;
    phoneNumber?: string;
    dateOfBirth?: string;
    address?: string;
    city?: string;
    avatarUrl?: string;
    isActive: boolean;
    isEmailVerified: boolean;
    themePreference?: string;
    languagePreference?: string;
    createdAt: string;
    updatedAt?: string;
}
